RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
FWIW the following code also shows up as a false positive (bug?) in the ObjectVariableNotSet inspection:

```
Private Sub TestEnum()
    Dim enumVariable As VBA.VbMsgBoxResult
    enumVariable = VBA.VbMsgBoxResult.vbIgnore
End Sub
```
The inspection believes the above assignment should use "Set".

So, I'm wondering if I should work on the ObjectVariableNotSet inspection  along with the bug for this QuickFix since they seem to be close cousins.(sound OK?)  And, assuming it is OK, is combing the inspection work under this same PR OK - or would you rather have any changes to the inspection submitted as a separate PR?